frontend/src/components/CalcSpectrum.tsx

```diff
@@ -41,6 +42,7 @@ export const CalcSpectrum: React.FC = () => {
   const [useGesia, setUseGesia] = useState(false);
   const [useSlit, setUseSlit] = useState(false); // checking that user wants to apply the slit function or not in available modes
   const [useSimulateSlitFunction, setUseSimulateSlitFunction] = useState(false); // checking the mode and enable or disable slit feature
+  const [downloadbutton, setDownloadbutton] = useState(false);
```

Comment:
```suggestion
  const [downloadButton, setDownloadButton] = useState(false);
```
since "download button" is two words, should be like this to follow camelCase

Reply:
I also think that since this is a boolean state, maybe a variable name like "showDownloadButton" would be more descriptive of what it is for?